Sources/Shared/Models/VideoTranscodeStatus.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  VideoTranscodeStatus.swift
+//  VimeoNetworking
+//
+//  Created by Nicole Lehrer on 12/9/19.
+//  Copyright © 2019 Vimeo. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+/// Representation of a video's transcoding status
+public struct VideoTranscodeStatus: Decodable {
+    
+    /// Current state of transcoding
+    var state: TranscodeState
+    
+    /// Percentage of transcoding completed
+    var progress: Int
+    
+    /// Time in seconds remaining until completion
+    var timeLeft: Double
```

Comment:
if this is in seconds should it just be an `int`?

Reply:
When measuring background upload duration to report to fresnel, I have encountered issues client side where time goes on forever and returns a huge value, causing overflow. I assumed it might be best to safeguard against this case if transcoding happens to hang for a very long time.

Reply:
`Int.max` in 64bits will be `18446744073709551615` seconds - do you think it is worth it making this into a `Double` just for that edge case? Then we have to deal with all the difficulties associated with that type i.e. I can see us having to cast this to `Int` to use with `TimeInterval`, comparisons not being possible etc.

I don't feel super strongly about it but from a consumer of this type I'd definitely prefer it to be an `Int` instead.

Reply:
Given we won't crash at run time for this model, and can just catch an error, it's not worth using a sub-optimal type for edge cases. Fair point, down to update to `Int`. 

Just checking for my knowledge, Isn't `TimeInterval` typealias for `Double`?

Reply:
Wow who knew I always thought it was an `Int`! Given that I think it is actually fine to leave it as a double if you want to revert it, sorry about that!